apps/backend/src/collectivites/discussions/infrastructure/discussion.repository.impl.ts

```diff
@@ -0,0 +1,335 @@
+  ): Promise<Result<DiscussionWithActionName[]>> {
+    try {
+      const conditions: (SQL | SQLWrapper | undefined)[] = [
+        eq(discussionTable.collectiviteId, collectiviteId),
+        like(discussionTable.actionId, `${referentielId}%`),
+      ];
+
+      // Apply filters
+      if (filters?.status) {
+        conditions.push(eq(discussionTable.status, filters.status));
+      }
+
+      if (filters?.actionId) {
+        // In order to find all sub actions, we need to use a like query with a wildcard
+        conditions.push(like(discussionTable.actionId, `${filters.actionId}%`));
+      }
+
+      const query = this.databaseService.db
+        .select({
+          ...getTableColumns(discussionTable),
+          actionNom: sql<string>`${actionDefinitionTable.nom}`.as('actionNom'),
+          actionIdentifiant:
+            sql<string>`${actionDefinitionTable.identifiant}`.as(
+              'actionIdentifiant'
+            ),
+        })
+        .from(discussionTable)
+        .leftJoin(
+          actionDefinitionTable,
+          eq(discussionTable.actionId, actionDefinitionTable.actionId)
+        )
+        .where(and(...conditions));
+
+      // Apply sorting
+      if (options?.sort) {
+        options.sort.forEach((sort) => {
+          const column =
+            sort.field === 'actionId'
+              ? actionDefinitionTable.actionId
+              : sort.field === 'created_at'
+              ? discussionTable.createdAt
+              : discussionTable.status;
+
+          query.orderBy(sort.direction === 'asc' ? column : desc(column));
+        });
+      }
+
+      // Apply pagination
+      if (options && 'limit' in options && options.limit !== 'all') {
+        if ('page' in options) {
+          query.limit(options.limit).offset((options.page - 1) * options.limit);
+        }
+      }
+
+      const discussions = await query;
+
+      return { success: true, data: discussions };
+    } catch (error) {
+      this.logger.error(
+        `Error listing discussions for collectivite ${collectiviteId} and referentiel ${referentielId}: ${error}`
+      );
+      return {
+        success: false,
+        error: DiscussionErrorEnum.DATABASE_ERROR,
+      };
+    }
+  }
+  create: (
+    discussion: CreateDiscussionData,
+    tx?: Transaction
+  ) => Promise<Result<DiscussionType>> = async (
+    discussion,
+    tx
+  ): Promise<Result<DiscussionType>> => {
+    try {
+      const discussionToCreate: CreateDiscussionType = {
+        collectiviteId: discussion.collectiviteId,
+        actionId: discussion.actionId,
+        status: DiscussionStatutEnum.OUVERT,
+        createdBy: discussion.createdBy,
+        createdAt: new Date().toISOString(),
+        modifiedAt: new Date().toISOString(),
+      };
+      const result = await (tx ?? this.databaseService.db)
+        .insert(discussionTable)
+        .values(discussionToCreate)
+        .returning();
+      if (!result || result.length === 0) {
+        return {
```

Comment:
ici un log serait pertinent:

 this.logger.error('INSERT RETURNING returned no result', {
        operation: 'create_discussion',
        collectiviteId: discussion.collectiviteId,
        actionId: discussion.actionId,
        createdBy: discussion.createdBy,
        hasTransaction: !!tx,
        timestamp: new Date().toISOString(),
      });